docs/detailed-push-process.md

```diff
@@ -5,53 +5,49 @@ But while doing so, it makes sure those components are deployed correctly and wo
+The numbers on the various arrows on the image indicate the order of the various steps.
 
 ## 1. Uploading the code
 
+When you run the `epinio push` command, the first thing the cli is going to do, is to put your code inside a tarball and send it to the `upload` endpoint of the Epinio API server which is running inside the Kubernetes cluster.
+
+## 2. Pushing the code to Gitea
+
 One of the components Epinio installs on your cluster is [Gitea](https://gitea.io/en-us/). Gitea is an Open Source code hosting solution. Among other things it allows
-us to create repositories and organizations using API calls. It is also used to store your application's code which Epinio pushes using [`git`](https://git-scm.com/).
+us to create repositories using API calls. Those repositories are used to store your application's code.
+
+So the first thing the Epinio API server does when it receives the upload request (the previous step), is to create a new project on Gitea. Then it uses `git` to push your code there.
 
-So the first thing Epinio does when you push your application for the first time is to create a new project on Gitea and uses `git` to push your code there.
-This doesn't mean you should be using `git` yourself. Epinio will create a tmp directory which will be the local git repository, copy your code over and then
-commit all the local changes you may have (even if you haven't commited those yet on your own git branch).
-Then it will push your code to Gitea.
+## 3. Staging the app
 
-## 2. Trigger of the Tekton pipeline
+When the upload request is complete, the cli will send a request to the `stage` endpoint of the Epinio API server. This will instruct the server to start the staging of the uploaded code.
 
-Now Gitea has your code but we need to tell buildpacks to build an image for your application. The buildpacks community maintains
-[a set of Tekton pipeline definitions](https://github.com/tektoncd/catalog/tree/main/task/buildpacks) that can be used to do exactly that.
-All we need to do is to tell the pipeline [where to find the application source](https://github.com/epinio/epinio/blob/bdcb1833829d58c5bdcf0e8ecb1998d766e72a13/embedded-files/tekton/triggers.yaml#L36-L44)
-and then trigger the pipeline to start.
+## 4. Trigger the pipeline
 
-Triggering happens using a [TriggerBinding](https://github.com/tektoncd/triggers/blob/main/docs/triggerbindings.md) to which Gitea will make a request.
-When you pushed your application in the previous step, Epinio set up a webhook on the Gitea project that makes a call to the trigger binding endpoint whenever
-the code is pushed to Gitea.
+When the Epinio API server receives the stage request, it will create a [`PipelineRun`](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md) that will run the staging Tekton pipeline using the version of the code referenced in the request. This pipeline has 3 steps. Their role is described in the following 3 steps.
```

Comment:
`has 3 steps ... following 3 steps.` might be better phrased as
`has 3 steps ... following 3 sections.`